src/Cameras/Inputs/freeCameraMouseWheelInput.ts

```diff
@@ -0,0 +1,373 @@
+import { Nullable } from "../../types";
+import { serialize } from "../../Misc/decorators";
+import { FreeCamera } from "../../Cameras/freeCamera";
+import { CameraInputTypes } from "../../Cameras/cameraInputsManager";
+import { BaseCameraMouseWheelInput } from "../../Cameras/Inputs/BaseCameraMouseWheelInput";
+import { Matrix, Vector3 } from "../../Maths/math.vector";
+
+/**
+ * Defines the potential axis to be altered in a transform operation.
+ */
+export enum AXIS {
+    /**
+     * X axis.
+     */
+    X,
+    /**
+     * Y axis.
```

Comment:
I feel like the naming of this may be a bit confusing as there is already an "Axis" class that has the same options (X, Y, Z) and returns a vector with corresponding values.

Reply:
Glad you brought this up.
There is already a `JOYSTICKAXIS`. See: https://doc.babylonjs.com/api/enums/babylon.joystickaxis
I considered creating a `MOUSEEWHEELAXIS` in a similar vein.... but was a bit concerned about namespace pollution.

I wondered if a generic `INPUTAXIS` might be more appropriate; It could then be reused for other uses. We'd probably want to put it in it's own file that we could include from here.

But such an enum would be potentially useful elsewhere (outside inputs) too.
But like you say, we'd then need to think of a better name for it.

Reply:
Actually,
I was confusing this with some other code i wrote recently.
Looking back at this PR,
the axis in question here isn't the mouse-wheel axis but rather the axis of the property the mouse wheel will affect.

Similar arguments could be made about re-usability vs polluting the namespace though.
With Babylon's commitment to backwards compatibility, the number of names in use is only going to grow...

Whatever we use has to be `export`-ed as it's used to configure the feature.

Can we come up with something that is reusable?
Do you have an official style-guide? This is the closest i found:
https://doc.babylonjs.com/how_to/contribute_to_api#format-of-comments
It implies enum names should actually PascalCase.
So what about:
`AxisEnum` ?
`EAxis` ?
`ChooseAxis` ?
`SelectAxis` ?

I'm not loving any of those.
Am i overthinking this?

Reply:
Just wondering (I'll review this better tomorrow) why not use the Axis enum we already have? It's rather genetic and has x,y and z

Reply:
There's a Class `Axis` here: `src/Maths/math.axis.ts`
and an Enum `JojstickAxis` here: `src/Misc/virtualJoystick.ts`.
I don't think i'm missing a plain Enum.

What about creating an Enum `Coordinate` in `src/Misc`?
It makes lexical sense and isn't too long.
`Cartesian` is another contender but it does not imply a single element but rather all 3 (X, Y and Z).

Naming is hard.

Reply:
Sorry, you are totally right.

The best would be to have an Axis enum, but... we already have an axis object. Naming... Well, Coordinate enum sounds good :-)

Reply:
Sold.
I've put it in `src/Maths/math.axis.ts` as it seems to fit there.